t/local/http.t

```diff
@@ -1,498 +1,475 @@
-#print STDERR $res->as_string;
-ok($res->is_success);
-
-# Then illegal credentials
-$ua->credentials($req->uri->host_port, "libwww-perl-digest", "user2", "passwd");
-$res = $ua->request($req);
-is($res->code, 401, 'response code 401');
-
-
-
-#----------------------------------------------------------------
-print "Check proxy...\n";
-sub httpd_get_proxy
 {
-   my($c,$r) = @_;
-   if ($r->method eq "GET" and
-       $r->uri->scheme eq "ftp") {
-       $c->send_basic_header(200);
-       $c->send_crlf;
-   }
-   else {
-       $c->send_error;
-   }
-}
-
-$ua->proxy(ftp => $base);
-$req = HTTP::Request->new(GET => "ftp://ftp.perl.com/proxy");
-$res = $ua->request($req);
-#print $res->as_string;
-ok($res->is_success);
-
-#----------------------------------------------------------------
-print "Check POSTing...\n";
-sub httpd_post_echo
-{
-   my($c,$r) = @_;
-   $c->send_basic_header;
-   $c->print("Content-Type: text/plain");
-   $c->send_crlf;
-   $c->send_crlf;
-
-   # Do it the hard way to test the send_file
-   open(TMP, ">tmp$$") || die;
-   binmode(TMP);
-   print TMP $r->as_string;
-   close(TMP) || die;
-
-   $c->send_file("tmp$$");
-
-   unlink("tmp$$");
+    package MyUA2;
+    use base 'LWP::UserAgent';
+    sub get_basic_credentials {
+        my($self, $realm, $uri, $proxy) = @_;
+        if ($realm eq "libwww-perl-digest" && $uri->rel($base) eq "digest") {
+            return ("ok 23", "xyzzy");
+        }
+        return undef;
+    }
 }
+sub daemonize {
+    my %router;
+    $router{delete_echo} = sub {
+        my($c, $req) = @_;
+        $c->send_basic_header(200);
+        $c->print("Content-Type: message/http\015\012");
+        $c->send_crlf;
+        $c->print($req->as_string);
+    };
+    $router{get_basic} = sub {
+        my($c, $r) = @_;
+        #print STDERR $r->as_string;
+        my($u,$p) = $r->authorization_basic;
+        if (defined($u) && $u eq 'ok 12' && $p eq 'xyzzy') {
+            $c->send_basic_header(200);
+            $c->print("Content-Type: text/plain");
+            $c->send_crlf;
+            $c->send_crlf;
+            $c->print("$u\n");
+        }
+        else {
+            $c->send_basic_header(401);
+            $c->print("WWW-Authenticate: Basic realm=\"libwww-perl\"\015\012");
+            $c->send_crlf;
+        }
+    };
+    $router{get_digest} = sub {
+        my($c, $r) = @_;
+        # print STDERR $r->as_string;
+        my $auth = $r->authorization;
+        my %auth_params;
+        if ( defined($auth) && $auth =~ /^Digest\s+(.*)$/ ) {
+            %auth_params = map { split /=/ } split /,\s*/, $1;
+        }
+        if ( %auth_params &&
+                $auth_params{username} eq "\"ok 23\"" &&
+                $auth_params{realm} eq "\"libwww-perl-digest\"" &&
+                $auth_params{qop} eq "auth" &&
+                $auth_params{algorithm} eq "\"MD5\"" &&
+                $auth_params{uri} eq "\"/digest\"" &&
+                $auth_params{nonce} eq "\"12345\"" &&
```

Comment:
Could we make the quotes more readable?

`$auth_params{nonce} eq q{"12345"} &&`